src/surface/Surface.ts

```diff
@@ -0,0 +1,156 @@
+/* eslint-disable no-useless-constructor */
+import { ButtonStyle, ILaunchpad, Style } from '../launchpads/base/ILaunchpad.js';
+import { allXy } from '../utils.js';
+import { ILayer } from './ILayer.js';
+import { Layer } from './Layer.js';
+
+/**
+ * Represents the visible Surface of a Launchpad
+ *
+ * Allows read/write access to the currently displayed buttons
+ * on the Launchpad. At any point, the in-memory display buffer can
+ * be written to the Launchpad. Commands will only be sent to
+ * render the difference.
+ *
+ * Supports multiple layers, so that it is easy to show temporary colors
+ * (for example, while a button is being held down) and return to the
+ * original color unchanged afterwards. Setting colors on the Surface
+ * directly is the same as setting them on layer 0.
+ *
+ * Note that on higher layers, palette color 0 and 'off' are not the same: 'off'
+ * will make the underlying button show through, while palette color 0 will make
+ * the button actively turn off.
+ *
+ * There should only be one surface in use for any Launchpad
+ * at a given time.
+ */
+export class Surface implements ILayer {
+  public readonly width = 9;
```

Comment:
What about the launchpad pro? Or do we not care about that atm

Reply:
I don't know 🤣 don't have one...

Reply:
Once we support one, it would be easy enough to add `.width` and `.height` to `ILaunchpad`, and have `Surface` initialize its own size from those members.

This is the only place those numbers are hardcoded, so the rest of the code should just work out.